htslib/sam.h

```diff
@@ -1367,6 +1557,27 @@ const char *sam_parse_region(sam_hdr_t *h, const char *s, int *tid,
 HTSLIB_EXPORT
 uint8_t *bam_aux_get(const bam1_t *b, const char tag[2]);
 
+/// Return a SAM formatting string containing a BAM tag
+/** @param b   Pointer to the bam record
+    @param tag Desired aux tag
+    @param s   The kstring to write to.
+
+    @return The number of bytes written to s on success, or -1 on error.
+    A return value of zero means no tag was found and errno is set toENOENT.
+ */
+static inline ssize_t bam_aux_get_str(const bam1_t *b,
+                                      const char tag[2],
+                                      kstring_t *s) {
+    const uint8_t *t = bam_aux_get(b, tag);
+    if (!t) return 0;
+
+    size_t len = s->l;
+    if (!sam_format_aux1(t-2, *t, t+1, b->data + b->l_data, s))
+        return -1;
+
+    return s->l - len;
```

Comment:
Could we make this return 1 when it adds a tag?  `ssize_t` causes problems with MSVC and while it is used in other htslib headers, so MSVC users have to deal with it already, this would be the first use in `htslib/sam.h`.  Returning 1 would also simplify the function a bit (no need for `len`) and I don't see that much utility in returning how many bytes were added.